clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2888,43 +2895,49 @@ class SYCLKernelNameTypeVisitor
       if (KernelNameIsMissing) {
         S.Diag(KernelInvocationFuncLoc, diag::err_sycl_kernel_incorrectly_named)
             << /* kernel name is missing */ 0;
+        IsInvalid = true;
       } else {
-        if (Tag->isCompleteDefinition())
+        if (Tag->isCompleteDefinition()) {
           S.Diag(KernelInvocationFuncLoc,
                  diag::err_sycl_kernel_incorrectly_named)
               << /* kernel name is not globally-visible */ 1;
-        else
+          IsInvalid = true;
+        } else
           S.Diag(KernelInvocationFuncLoc, diag::warn_sycl_implicit_decl);
 
         S.Diag(Tag->getSourceRange().getBegin(), diag::note_previous_decl)
             << Tag->getName();
       }
+      return isValid();
     }
+    return true;
   }
 
-  void VisitTypeTemplateArgument(const TemplateArgument &TA) {
+  bool VisitTypeTemplateArgument(const TemplateArgument &TA) {
     QualType T = TA.getAsType();
     if (const auto *ET = T->getAs<EnumType>())
-      VisitEnumType(ET);
+      return VisitEnumType(ET);
```

Comment:
Explicit "return" keyword was added here(and in few other places) so that the compiler wouldn't throw the following warning: `no return statement in function returning non-void [-Wreturn-type]`